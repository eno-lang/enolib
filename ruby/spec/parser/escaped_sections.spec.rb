# frozen_string_literal: true

input = <<~DOC.strip
# `key`
## ``long ` key``
    ### ```ke``y```
    ####    ` key`
DOC

describe Enolib::Parser do
  describe 'Escaped sections' do
    context = Enolib::Context.new(input)

    it 'parses as expected' do
      expect(context.document).to match_snapshot
    end
  end
end
