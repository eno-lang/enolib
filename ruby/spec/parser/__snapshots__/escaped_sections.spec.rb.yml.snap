---
"./spec/tokenizer/escaped_sections.spec.rb[1:1:1]":
- :index: 0
  :line: 0
  :depth: 1
  :type: :section
  :key: name
  :ranges:
    :escape_begin_operator:
    - 2
    - 3
    :escape_end_operator:
    - 7
    - 8
    :key:
    - 3
    - 7
    :section_operator:
    - 0
    - 1
  :length: 8
- :index: 9
  :line: 1
  :depth: 2
  :type: :section
  :key: long ` name
  :ranges:
    :escape_begin_operator:
    - 3
    - 5
    :escape_end_operator:
    - 16
    - 18
    :key:
    - 5
    - 16
    :section_operator:
    - 0
    - 2
  :length: 18
- :index: 28
  :line: 2
  :depth: 3
  :type: :section
  :key: na``me
  :ranges:
    :escape_begin_operator:
    - 8
    - 11
    :escape_end_operator:
    - 17
    - 20
    :key:
    - 11
    - 17
    :section_operator:
    - 4
    - 7
  :length: 20
- :index: 49
  :line: 3
  :depth: 4
  :type: :section
  :key: name
  :ranges:
    :escape_begin_operator:
    - 12
    - 13
    :escape_end_operator:
    - 18
    - 19
    :key:
    - 14
    - 18
    :section_operator:
    - 4
    - 8
  :length: 19
"./spec/parser/escaped_sections.spec.rb[1:1:1]": &4
  :elements:
  - &3
    :index: 0
    :line: 0
    :ranges:
      :line:
      - 0
      - 7
      :section_operator:
      - 0
      - 1
      :escape_begin_operator:
      - 2
      - 3
      :escape_end_operator:
      - 6
      - 7
      :key:
      - 3
      - 6
    :elements:
    - &2
      :index: 8
      :line: 1
      :ranges:
        :line:
        - 8
        - 25
        :section_operator:
        - 8
        - 10
        :escape_begin_operator:
        - 11
        - 13
        :escape_end_operator:
        - 23
        - 25
        :key:
        - 13
        - 23
      :elements:
      - &1
        :index: 26
        :line: 2
        :ranges:
          :line:
          - 26
          - 45
          :section_operator:
          - 30
          - 33
          :escape_begin_operator:
          - 34
          - 37
          :escape_end_operator:
          - 42
          - 45
          :key:
          - 37
          - 42
        :elements:
        - :index: 46
          :line: 3
          :ranges:
            :line:
            - 46
            - 64
            :section_operator:
            - 50
            - 54
            :escape_begin_operator:
            - 58
            - 59
            :escape_end_operator:
            - 63
            - 64
            :key:
            - 60
            - 63
          :elements: []
          :type: :section
          :parent: *1
          :key: key
        :type: :section
        :parent: *2
        :key: ke``y
      :type: :section
      :parent: *3
      :key: long ` key
    :type: :section
    :parent: *4
    :key: key
  :type: :document
